cli/pkg/kctrl/cmd/package/builder/create.go

```diff
@@ -0,0 +1,450 @@
+
+	ValuesSchema      bool
+	DefaultValuesFile string
+
+	pkgCmdTreeOpts cmdcore.PackageCommandTreeOpts
+}
+
+func NewCreateOptions(ui ui.UI, logger logger.Logger, pkgCmdTreeOpts cmdcore.PackageCommandTreeOpts) *CreateOptions {
+	return &CreateOptions{ui: ui, logger: logger, pkgCmdTreeOpts: pkgCmdTreeOpts}
+}
+
+func NewCreateCmd(o *CreateOptions) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "create",
+		Aliases: []string{"g"},
+		Short:   "Create a package",
+		Args:    cobra.ExactArgs(1),
+		RunE:    func(_ *cobra.Command, args []string) error { return o.Run(args) },
+		Example: cmdcore.Examples{
+			cmdcore.Example{"Create a package",
+				[]string{"package", "builder", "create"},
+			},
+		}.Description("", o.pkgCmdTreeOpts),
+		SilenceUsage: true,
+		Annotations:  map[string]string{"table": ""},
+	}
+
+	return cmd
+}
+
+func (o *CreateOptions) Run(args []string) error {
+	//TODO Rohit Should we provide an option to give pkg location?
+	pkgLocation := GetPkgLocation()
+	createStep := NewCreateStep(o.ui, pkgLocation)
+	err := createStep.Run()
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+type CreateStep struct {
+	Ui           ui.UI
+	PkgLocation  string
+	PkgVersion   string
+	FqName       string
+	ValuesSchema v1alpha1.ValuesSchema
+	Fetch        fetch.FetchStep
+	Template     template.TemplateStep
+	Maintainers  []v1alpha1.Maintainer
+}
+
+func NewCreateStep(ui ui.UI, pkgLocation string) *CreateStep {
+	return &CreateStep{
+		Ui:          ui,
+		PkgLocation: pkgLocation,
+	}
+}
+
+func (createStep *CreateStep) Run() error {
+	err := createStep.PreInteract()
+	if err != nil {
+		return err
+	}
+	err = createStep.Interact()
+	if err != nil {
+		return err
+	}
+	err = createStep.PostInteract()
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (createStep CreateStep) getStartBlock() []byte {
+	str := fmt.Sprintf(`
+# Lets start on the package creation process.
+# First we need a directory to store all configurations
+# Creating directory %s
+# 	$ mkdir -p %s
+`, createStep.PkgLocation, createStep.PkgLocation)
+	return []byte(str)
+}
+
+func (create CreateStep) PreInteract() error {
+	create.Ui.PrintBlock(create.getStartBlock())
+	output, err := util.Execute("mkdir", []string{"-p", create.PkgLocation})
+	if err != nil {
+		create.Ui.ErrorLinef("Error creating package directory.Error is: %s", err.Error())
+		return err
+	}
+	create.Ui.PrintBlock([]byte(output))
+	return nil
+}
+
+func (create *CreateStep) getPkgVersionBlock() []byte {
+	str := `
+# A package can have multiple versions. These versions are used by PackageInstall to install specific version of the package into the Kubernetes cluster.`
+	return []byte(str)
+}
+
+func (create *CreateStep) getFQPkgNameBlock() []byte {
+	str := `# A package name must be a fully qualified name. 
+# It must consist of at least three segments separated by a '.'
+# Fully Qualified Name cannot have a trailing '.' e.g. samplePackage.corp.com`
+	return []byte(str)
```

Comment:
Right now, after entering the package version, we are asked for package name, the output is something like this
```bash
Enter the package version (): 1.0.0

# A package name must be a fully qualified name. 
# It must consist of at least three segments separated by a '.'
# Fully Qualified Name cannot have a trailing '.' e.g. samplePackage.corp.com
Enter the fully qualified package name ():
```

So, as soon as we enter the version, we see a text saying `# A package name must be a fully qualified name.` and it seems like an error at first.

Maybe we can simply ask for the package name with an example, and if user provides an invalid name then provide the instructions? 
Something like this:
```bash
Enter the fully qualified package name (e.g. sample-package.corp.com):  foo

# A package name must be a fully qualified name. 
# It must consist of at least three segments separated by a '.'
# Fully Qualified Name cannot have a trailing '.'

Enter the fully qualified package name (e.g. sample-package.corp.com): my-package.open.ai
```

Reply:
Same for all inputs.

Reply:
I think asking version and then FQ name is a bit disconnected and looks like anti flow. We should ask FQ name and then version. What do you think ?

Reply:
Yep, I was suggesting the same [here](https://github.com/vmware-tanzu/carvel-kapp-controller/pull/634#discussion_r860707626) xD

Reply:
We talked about it and agreed that this is part of educational information, hence t needed to be added.